### ISSUE TITLE
import of custom csv + signals (from closed source indicators)

### ISSUE THREAD

**Issue by luboremo** [freqtrade/freqtrade#3111]
Is it somehow possible to import custom signal data to freqtrade?
I use some closed source indicators on Tradingview, from there I can export csv files with UNIX timestamp and OHLCV + signals from the indicators.

So I could create example.json with data structure like: ['date', 'open', ''high', 'low', 'close', 'volume', 'signal1', 'signal2', 'signal3']

Where signal can be just 0,1, or empty.

Basically I need to write a strategy that will just take signals from this document when backtesting. Would it be possible somehow with freqtrade?

**Comment by xmatthias** [comment]
well the strategy is just a python file - so with some python / pandas skills i assume you can load a CSV file, mege that to the dataframe freqtrade provides (based on date, most likely) - and use the "signal" columns from there.

It's not something supported out of the box - and not something we'll be helping you with to implement in detail, other than pointing you in the right direction.


Also i'm not sure on the benefits/reason for doing this if i'm honest - you'd do backtesting, OK - but since it's closed-source indicators, you'd not be able to use them for real trading anyway.

**Comment by luboremo** [comment]
thanks for the hint, I will try to do it that way. I just wanted to know if it would even be possible with freqtrade since I'm not a coder, just learning by doing = a lot of Google and Stackoverflow.

I have access to the indicator outputs (not the source code) + data export to .csv, but I am looking for backtesting tool that can help me to backtest the strategy across various pairs + in combination with other indicators (to avoid backtesting it manually and make the backtesting more efficient).

**Comment by hroff-1902** [comment]
> just learning by doing = a lot of Google and Stackoverflow.

and this is the right direction 👍

**Comment by lukacsaron** [comment]
@luboremo any chance you succeeded here? I'm struggling with the same here trying to backtest based on signal data from a csv
